### PR TITLE
User could provide log validator for game-loop test. Enabled retry logic for failed tests.

### PR DESCRIPTION
**Pull request by sunmou99**
1. User could provide log validator for game-loop test. 
2. Enabled retry logic for failed tests.
3. Added README.md

Test example: rerun firestore test after failure occurs:
https://github.com/firebase/firebase-cpp-sdk/runs/8123763179?check_suite_focus=true#step:12:4341

### CODE REVIEW

File: firebase-test-lab/trigger_ftl_tests.py [hunk 1]
```diff
@@ -81,6 +81,7 @@
 import argparse
 import logging
 import json
+import imp
```
Comment (yifanyang) [inline firebase-test-lab/trigger_ftl_tests.py:84]: `imp` is deprecated in favor of `importlib`.

File: firebase-test-lab/README.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    with:
+      ...
+- run: echo '${{ steps.ftl_test.outputs.test_summary }}'
+```
+
+Test summary example:
+```
+{
+  "project_id": ${project_id},
+  "apps": [
```
Comment (yifanyang) [inline firebase-test-lab/README.md:150]: How about calling it "attempts" or "executions"?

File: firebase-test-lab/trigger_ftl_tests.py [hunk 3]
```diff
@@ -185,67 +186,95 @@ def _run_test_on_ftl(FLAGS):
...
-    "ftl_link": ftl_link,
-    "raw_result_link":  raw_result_link,
-    "devices": outcome_device
-  }
-  tests_result.get('apps').append(test_summary)
+  if FLAGS.test_type == GAMELOOP and FLAGS.validator:
+    try:
+      module = os.path.splitext(FLAGS.validator)[0]
+      validator = imp.load_source(module, FLAGS.validator)
+      return validator.validate(test_summary)
```
Comment (yifanyang) [inline firebase-test-lab/trigger_ftl_tests.py:277]: Users will need the class/interface definition for validator and test_summary in order to provide their implementation. Do you want to clarify that?

File: firebase-test-lab/trigger_ftl_tests.py (hunk 1 above)
Comment (sunmou99) [reply firebase-test-lab/trigger_ftl_tests.py:84]: This is for Python 2 users.

File: firebase-test-lab/trigger_ftl_tests.py (hunk 3 above)
Comment (sunmou99) [reply firebase-test-lab/trigger_ftl_tests.py:277]: I don't think so, and I didn't mention it in README too.
As we talked yesterday, the validation process is unique to game-loop test and games team is the main user if not the only user.
If there are requirements comes for other users in the future, we could have a better design then.

Comment (yifanyang) [reply firebase-test-lab/trigger_ftl_tests.py:277]: It is exposed as an input option in the action yaml, isn't it? Still it's up to you.

Comment (sunmou99) [reply firebase-test-lab/trigger_ftl_tests.py:277]: I think it's fine, I also added "[Experimental]" to the input description.
If you have any ideas on how to hide this input from action yaml, please let me know.

**Review by yifanyang — APPROVED** [review]

**Pull request merged**